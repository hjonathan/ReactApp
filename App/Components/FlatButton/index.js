import React, {Component} from "react";
import {
    Text,
    TouchableOpacity,
    View
} from "react-native";
import Styles from "./styles";

/**
 * Class FlatButton
 * Example:
 *
 * import FlatButton from '../FlatButton';
 * render() {
 *      return (
 *          <FlatButton
 *              params={{radius: 4, border: 1, color: '#fff'}}
 *              onPress={() => {}}
 *              title="Title test"
 *          />
 *      );
 * }
 */
export default class Button extends Component {
    /**
     * Constructor FlatButton
     * @param props
     */
    constructor (props) {
        super(props);
        this.state = {
            title: (props && props.title) || "test"
        };
    }

    /**
     * Returns custom button
     * @param params
     * @returns {*}
     */
    renderCustomButton (params) {
        return (
            <View style={{
                borderRadius: params.radius,
                borderWidth: params.border,
                borderColor: params.color,
                alignItems: "center",
                paddingHorizontal: 50,
                paddingVertical: 10
            }}
            >
                <Text style={{color: params.color, fontWeight: "bold"}}>
                    {this.state.title}
                </Text>
            </View>
        );
    }

    /**
     * Returns default button
     * @returns {*}
     */
    renderDefaultButton () {
        return (
            <View style={Styles.button}>
                <Text style={Styles.text}>
                    {this.state.title}
                </Text>
            </View>
        );
    }

    /**
     * Renders Component
     * @param onPress
     * @returns {*}
     */
    render ({onPress} = this.props) {
        return (
            <TouchableOpacity onPress={onPress}>
                {
                    (Object.prototype.hasOwnProperty.call(this.props, "params") && this.props.params)
                        ? this.renderCustomButton(this.props.params) : this.renderDefaultButton()
                }
            </TouchableOpacity>
        );
    }
}
