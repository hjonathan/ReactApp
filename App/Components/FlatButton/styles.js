import {StyleSheet} from "react-native";

export default StyleSheet.create({
    button: {
        alignItems: "center",
        borderRadius: 50,
        borderWidth: 2,
        borderColor: "#FFFFFF",
        paddingHorizontal: 50,
        paddingVertical: 10
    },
    text: {
        color: "#FFFFFF",
        fontWeight: "bold"
    }
});
