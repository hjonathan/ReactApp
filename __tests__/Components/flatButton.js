import 'react-native';
import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import FlatButton from '../../App/Components/FlatButton';

describe('Component >>> FlatButton', () => {
    let defaultFlatButton,
        customFlatButton;

    const props = {
        params: {
            radius: 5,
            border: 1,
            color: "red"
        },
        onPress: () => {}
    };

    beforeEach(() => {
        defaultFlatButton = shallow(<FlatButton params={props.params} onPress={props.onPress} />);
        customFlatButton = shallow(<FlatButton title={""} onPress={props.onPress} />);
    });

    it('Render DefaultFlatButton', () => {
        expect(toJson(defaultFlatButton)).toMatchSnapshot();
    });

    it('Render CustomFlatButton', () => {
        expect(toJson(customFlatButton)).toMatchSnapshot();
    });
});
