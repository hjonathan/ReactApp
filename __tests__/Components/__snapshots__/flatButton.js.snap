// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component >>> FlatButton Render CustomFlatButton 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#FFFFFF",
        "borderRadius": 50,
        "borderWidth": 2,
        "paddingHorizontal": 50,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFFFFF",
          "fontWeight": "bold",
        }
      }
    >
      test
    </Text>
  </Component>
</TouchableOpacity>
`;

exports[`Component >>> FlatButton Render DefaultFlatButton 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "borderColor": "red",
        "borderRadius": 5,
        "borderWidth": 1,
        "paddingHorizontal": 50,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "red",
          "fontWeight": "bold",
        }
      }
    >
      test
    </Text>
  </Component>
</TouchableOpacity>
`;
